Classes/Controller/MediaController.php

```diff
@@ -45,8 +49,26 @@ public function videoAction() {
         $langId = $context->getPropertyFromAspect('language', 'id');
         $site = $GLOBALS['TYPO3_REQUEST']->getAttribute('site');
         $language = $site->getLanguageById($langId);
-        $langCode = $language->getTwoLetterIsoCode();
-        $this->view->assign('langCode', $langCode);
+        $locale = $language->getTwoLetterIsoCode();
```

Comment:
Actually this is not the way you should access the language in TYPO3 as far as I can see it.

Your example works somehow, but the player is still on a german website. 

Thing to do in DDEV system:
* List-module: Create a website language record in "root" (UID=0)
* Page-module: Create a "translation of this page" on page "Werkansicht"
* Page-module: Translate content using the wizard.

Now Fluid will render all translation in the proper language. I would prefer to let Fluid translate the labels and not to do this in the controller. We've done this e.g. in https://github.com/slub/geospatial-search-frontend (JavaScript-App) together with slub_web_digas (SLUB Gitlab). That's maybe not the perfect solution and lacks the support for plurals (good idea!) but maybe it's worth to have a look on it.

![Sachsen Digital  TYPO3 CMS 9 5 31 ](https://user-images.githubusercontent.com/3354500/140929423-71d159ed-cb49-4b54-b886-7e4664f60ade.png)

![Sachsen Digital  TYPO3 CMS 9 5 31  (1)](https://user-images.githubusercontent.com/3354500/140929449-7eb8cf54-6559-4d91-84c3-fa03f62c93a6.png)

Reply:
To clarify, the language code is accessed for passing it to the Shaka player (which brings its own translations), and for selecting pluralization rules in the JS part. So because we probably want Shaka to be localized, I don't see a way around accessing the language code (or is there some other/better way to do this?) The code is not used for actually translating the labels. (Previously I had used it in the call to `getParsedData`, which isn't necessary because I only use the `'default'` key from the result.)

Reply:
Could you explain why you prefer to use Fluid for translating the labels? (I'm not having a firm opinion on how to implement this, so I'm curious about considerations/concerns.)

Compared to `slub_web_digas`, there's two differences:
- I'm collecting all available labels / keys from the default language file `locallang_video.xlf` to avoid manually listing them. Do you think there's an issue with this?
- As you mentioned, I'm not using the Fluid viewhelper. In fact, from the code (in TYPO3 v9.5, but also v11.3 and master) it looks like the viewhelper also calls `LocalizationUtility::translate()` and pipes the result to `vsprintf()`. (And the result is escaped, which we'd probably want to bypass.)

If it's about caching, another option could be to move the code from the controller to a separate viewhelper.